data/mods/Craft_Gunpowder/cgp_recipes.json

```diff
@@ -238,8 +246,9 @@
   ], "components": [
     [
       [ "pipe", 3 ]
-    ],
+    [
```

Comment:
How is that supposed to work? Have you tested this? You should at least start a new game in a world that includes the mod. The game will report errors like this when it loads the world data.